app/ui/src/main/java/de/mm20/launcher2/ui/launcher/search/location/LocationItem.kt

```diff
@@ -0,0 +1,306 @@
+
+                    Toolbar(
+                        leftActions = listOf(
+                            DefaultToolbarAction(
+                                label = stringResource(id = R.string.menu_back),
+                                icon = Icons.Rounded.ArrowBack
+                            ) {
+                                onBack()
+                            }
+                        ),
+                        rightActions = listOfNotNull(
+                            DefaultToolbarAction(
+                                label = stringResource(id = R.string.menu_map),
+                                icon = Icons.Rounded.Map
+                            ) {
+                                viewModel.viewModelScope.launch {
+                                    location.launch(context, null)
+                                }
+                            },
+                            location.websiteUrl.runCatching {
+                                val uri = Uri.parse(this)
+                                DefaultToolbarAction(
+                                    label = stringResource(id = R.string.menu_website),
+                                    icon = Icons.Rounded.TravelExplore
+                                ) {
+                                    viewModel.viewModelScope.launch {
+                                        context.startActivity(Intent(Intent.ACTION_VIEW, uri))
+                                    }
+                                }
+                            }.getOrNull(),
+                        )
+                    )
+
+                }
+
+
+            }
+        }
+    }
+}
+
+private fun Location.getSummary(context: Context, distance: Float?): String {
+    val summary = StringBuilder()
+    if (distance != null) {
+        summary.append(distance.metersToString(context), ' ')
+    }
+    if (this.street != null) {
+        summary.append(this.street, ' ')
+        if (this.houseNumber != null) {
+            summary.append(this.houseNumber, ' ')
+        }
+    }
+    return summary.toString()
+}
+
+private fun Float.metersToString(context: Context): String {
+    val isKm = this >= 1000f
+    val value =
+        if (isKm) DecimalFormat().apply { maximumFractionDigits = 1; minimumFractionDigits = 0 }
+            .format(this / 1000f)
+        else this.roundToInt().toString()
+    val unit =
+        context.getString(if (isKm) R.string.unit_kilometer_symbol else R.string.unit_meter_symbol)
+
+    return "$value $unit"
+}
+
+private fun LocationCategory?.getImageVector(): ImageVector = when (this) {
+    LocationCategory.RESTAURANT -> Icons.Rounded.Restaurant
+    LocationCategory.FAST_FOOD -> Icons.Rounded.Fastfood
+    LocationCategory.BAR -> Icons.Rounded.LocalBar
+    LocationCategory.CAFE -> Icons.Rounded.LocalCafe
+    LocationCategory.HOTEL -> Icons.Rounded.Hotel
+    LocationCategory.SUPERMARKET -> Icons.Rounded.LocalGroceryStore
+    else -> Icons.Rounded.Place
+}
```

Comment:
This should be moved to `Location::getPlaceholderIcon`.
This would allow
  1. us to use some color for the icons / use different colors for different categories
  2. icons to respect the user's themed icon preference
  3. users to customize the icons

Icons need to be provided as XML Vector Drawables because we don't have access to Compose icons at this level. See `File.kt` for an example.